code/modules/mob/abstract/new_player/sprite_accessories.dm

```diff
@@ -28,7 +28,7 @@
 	var/gender = NEUTER
 
 	// Restrict some styles to specific species
-	var/list/species_allowed = list("Human")
+	var/list/species_allowed = list("Human","Shell Frame")

```

Comment:
This might not be needed, as this var is based on the body type, and shells have the same body type as humans.

Reply:
The code as it stands seems to check for the species name rather than the bodytype. There's a lot involved and that's why I just resorted to adding in the species name.
Without Shell Frame on the species_allowed list, it doesn't have any valid hairstyles to choose from.

Reply:
I also suspect it's a similar reason why the character setup menu only picks the Bald hairstyle when using randomise body for Shell Frames.